components/constellation/constellation.rs

```diff
@@ -1020,6 +1023,23 @@ impl<Message, LTF, STF> Constellation<Message, LTF, STF>
         }
     }
 
+    fn handle_broadcast_storage_event(&self, pipeline_id: PipelineId, storage: StorageType, url: String,
+                                      key: Option<String>, old_value: Option<String>, new_value: Option<String>) {
+        let origin = match Url::parse(&*url) {
+            Ok(url) => url.origin(),
+            Err(err) => return warn!("Failed to parse url {}: {:?}.", url, err),
+        };
```

Comment:
<strike>Why ignore failure here? Why isn't that method taking an `Url` directly? I see below that the `String` comes directly from an `Url`, so please just unwrap the result of `Url::parse` or pass an actual `Url` around.</strike>

Oh I see you did that in second commit. Could you squash them?